### PR TITLE
Support Binary

### LINKED ISSUE #9: Support to binary data component
hello, it is necessary to perform the migration of the binary component (file upload) for this a gRPC-based service is configured that can send the data from the file to the server.

**ADempiere**
zk-binary-data

[**Element-UI**](https://element.eleme.io/#/en-US/component/upload#user-avatar-upload)
element-binary-data

**ADempiere-Vue Desktop**
adempiere-vue-binary-data

**ADempiere-Vue Mobile**
adempiere-vue-mobile-binary-data

### LINKED ISSUE #45: Suport to image component
Hello, it is proposed to migrate the Image component, this will support the data imported from gRPC and will conform to the ADempiere standards. Supported for JPG.PNG.BMP.GIF type image format, among others

- **ADempiere**
Peek 2019-11-18 12-40

- **Element-UI**
https://element.eleme.cn/#/en-US/component/upload

element

- **ADempiere-Vue**
Peek 2019-11-18 12-43
- **ADempiere-Vue (Mobile)**
Peek 2019-11-18 12-46

### PR DESCRIPTION
**Pull request by elsiosanchez**
## Bug report / Feature
Support to Binary Compunent

#### Steps to reproduce

1. Open a record
2. Upload an image
3. 
#### Screenshot or Gif

binary

#### Link to minimal reproduction

https://demo-ui.erpya.com/#/a48d2596-fb40-11e8-a479-7a0060f0aa01/0/window/113?tabParent=0

#### Other relevant information
- Your OS: Debian 9
- Web Browser: Chrome 
- Node.js version: 10.9.0

#### Issues:
Fixes: #9 
Fixes: #45

### CODE REVIEW

File: src/api/ADempiere/field/binary.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * @param {number} recordId
+ * @param {string} recordUuid
+ * @param {object} binary
+ * @param {array} attributesList
+ */
+export function requestGetBinary({
+  id,
+  uuid,
+  tableName,
+  attributesList
```
Comment (yamelsenih) [inline src/api/ADempiere/field/binary.js]: What is it?

Please change attributesList by specific attribute

**Review by yamelsenih — CHANGES_REQUESTED** [review]

File: src/api/ADempiere/field/binary.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+/**
+ * Get entity with binary by identifier
+ * @param {number} recordId
+ * @param {string} recordUuid
+ * @param {object} binary
+ */
+export function requestGetBinary({
```
Comment (yamelsenih) [inline src/api/ADempiere/field/binary.js]: Please change this by getResource

File: src/api/ADempiere/field/binary.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  id,
+  uuid,
+  tableName,
+  binary
```
Comment (yamelsenih) [inline src/api/ADempiere/field/binary.js]: What is the reason for binary?

File: src/api/ADempiere/field/binary.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+/**
+ * Update an existing binary by id or uuid
+ * @param {number} recordId
+ * @param {string} recordUuid
+ * @param {object} binary
+ * @param {array} attributesList
+ */
+export function requestUpdateBinary({
```
Comment (yamelsenih) [inline src/api/ADempiere/field/binary.js]: Please change this for uploadResource

File: src/api/ADempiere/field/binary.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  recordId: id,
+  recordUuid: uuid,
+  tableName,
+  attributesList
```
Comment (yamelsenih) [inline src/api/ADempiere/field/binary.js]: Is not good when you use generic object, change by real name

File: src/components/ADempiere/Field/FieldImage.vue [hunk 6]
```diff
@@ -1,24 +1,41 @@
...
-    :class="cssClassStyle"
-  >
-    <img v-if="value" :src="value" class="avatar">
-    <i v-else class="el-icon-plus avatar-uploader-icon" />
-  </el-upload>
+  <div>
+    {{ value }}
+    <el-upload
+      :ref="metadata.columnName"
+      :action="getImage()"
```
Comment (yamelsenih) [inline src/components/ADempiere/Field/FieldImage.vue]: I don't sure if this name is correct

File: src/components/ADempiere/Field/FieldImage.vue [hunk 7]
```diff
@@ -29,6 +46,15 @@ export default {
     }
   },
   methods: {
+    requestResourceReference,
+    imageServer() {
```
Comment (yamelsenih) [inline src/components/ADempiere/Field/FieldImage.vue]: ??? Please put a correct and descriptive ne

**Review by yamelsenih — CHANGES_REQUESTED** [review]
Please make this changes for test

**Comment by yamelsenih** [comment]
I tasted it and not working for mobile device 
See the gif

**Comment by elsiosanchez** [comment]
mobile phone support has been added

#### Screenshot or Gif

supportmovil

**Pull request merged**